app/policies/activity_policy.rb

```diff
@@ -5,35 +5,54 @@ def index?
 
   def show?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def create?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def update?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def destroy?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   private def fund?
     record.hierarchy.is_a?(Fund)
   end
 
+  private def programme?
+    record.hierarchy.is_a?(Programme)
+  end
+
+  private def fund_and_fund_manager?
+    fund? && user.fund_manager?
+  end
+
+  private def programme_and_fund_manager?
+    programme? && user.fund_manager?
+  end
+
   class Scope < Scope
     def resolve
       if user.administrator?
         scope.all
       else
+        # TODO: This will eventually become unsustainable,
+        # we need a better way to do this
         funds = Fund.where(organisation_id: user.organisation)
-        scope.where(hierarchy: funds)
+        programmes = Programme.where(fund_id: funds)
+        scope.where(hierarchy: [funds + programmes])
```

Comment:
Good point. Never considered that this scope will include all four levels, so we need to avoid a horrible query!